.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,51 @@
+name: Publish Python 🐍 distribution 📦 to PyPI
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
NIT: as a follow-up, it may be worth updating all re-usable actions here and in other workflows to be pinned by sha instead of tag, e.g.

```
    - uses: actions/checkout@1e31de5234b9f8995739874a8ce0492dc87873e2  # v4.0.0
```

which enables configuring dependabot to update the action versions automatically.

Reply:
@JRosenkranz Are we planning bot? Otherwise someone needs to manually update all SHAs to stay updated.